src/main/c/Objects/pyjclass.c

```diff
@@ -458,43 +473,21 @@ static PyGetSetDef pyjclass_getset[] = {
     {NULL} /* Sentinel */
 };
 
-PyTypeObject PyJClass_Type = {
-    PyVarObject_HEAD_INIT(NULL, 0)
-    "jep.PyJClass",
-    sizeof(PyJClassObject),
-    0,
-    (destructor) pyjclass_dealloc,            /* tp_dealloc */
-    0,                                        /* tp_print */
-    0,                                        /* tp_getattr */
-    0,                                        /* tp_setattr */
-    0,                                        /* tp_compare */
-    0,                                        /* tp_repr */
-    0,                                        /* tp_as_number */
-    0,                                        /* tp_as_sequence */
-    0,                                        /* tp_as_mapping */
-    0,                                        /* tp_hash  */
-    (ternaryfunc) pyjclass_call,              /* tp_call */
-    0,                                        /* tp_str */
-    pyjclass_getattro,                        /* tp_getattro */
-    pyjclass_setattro,                        /* tp_setattro */
-    0,                                        /* tp_as_buffer */
-    Py_TPFLAGS_DEFAULT,                       /* tp_flags */
-    "jclass",                                 /* tp_doc */
-    0,                                        /* tp_traverse */
-    0,                                        /* tp_clear */
-    0,                                        /* tp_richcompare */
-    0,                                        /* tp_weaklistoffset */
-    0,                                        /* tp_iter */
-    0,                                        /* tp_iternext */
-    pyjclass_methods,                         /* tp_methods */
-    pyjclass_members,                         /* tp_members */
-    pyjclass_getset,                          /* tp_getset */
-    0, // &PyJObject_Type                     /* tp_base */
-    0,                                        /* tp_dict */
-    0,                                        /* tp_descr_get */
-    0,                                        /* tp_descr_set */
-    offsetof(PyJClassObject, attr),           /* tp_dictoffset */
-    0,                                        /* tp_init */
-    0,                                        /* tp_alloc */
-    NULL,                                     /* tp_new */
+static PyType_Slot slots[] = {
+    {Py_tp_doc, "Jep java.lang.Class"},
+    {Py_tp_dealloc, pyjclass_dealloc},
+    {Py_tp_call, pyjclass_call},
+    {Py_tp_getattro, pyjclass_getattro},
+    {Py_tp_setattro, pyjclass_setattro},
+    {Py_tp_methods, pyjclass_methods},
+    {Py_tp_members, pyjclass_members},
+    {Py_tp_getset, pyjclass_getset},
+    {0, NULL},
+};
+
+PyType_Spec PyJClass_Spec = {
+    .name = "java.lang.Class",
```

Comment:
I take it this was intentional that you changed the name?  Will that affect anything?  I noticed below that PyJObject has java.lang.Object so this change makes it more consistent at least.

Reply:
It was not intentional, thanks for pointing it out. I am actually opposed to naming this java.lang.Class because class methods like getSuperclass() don't work on PyJClass so I think I should revert the name back to jep.PyJClass.